sklearn/decomposition/tests/test_truncated_svd.py

```diff
@@ -52,8 +52,11 @@ def test_too_many_components(algorithm, X_sparse):
     n_features = X_sparse.shape[1]
     for n_components in (n_features, n_features + 1):
         tsvd = TruncatedSVD(n_components=n_components, algorithm=algorithm)
-        with pytest.raises(ValueError):
+        if n_components == n_features and algorithm == "randomized":
```

Comment:
To actually test the new feature we can add to `test_attributes`:

```python
@pytest.mark.parametrize("n_components", (10, 25, 41, 55))
def test_attributes(n_components, X_sparse):
    n_features = X_sparse.shape[1]
    tsvd = TruncatedSVD(n_components).fit(X_sparse)
    assert tsvd.n_components == n_components
    assert tsvd.components_.shape == (n_components, n_features)
```

Reply:
Done.  Thank you for the feedback!